src/main/java/io/rcktapp/api/handler/sql/SqlDb.java

```diff
@@ -305,76 +310,123 @@ public void reflectDb() throws Exception
+                {
+                    String tableName = foreignKeyTablesRS.getString("TABLE_NAME");
+
+                    if (Stream.of(ignoreTablePrefixes.split(",")).filter(Predicate.not(String::isEmpty)).anyMatch(tableName::startsWith)) continue;
+
+                    keyFutures.add(CompletableFuture.supplyAsync(() -> {
+                        try (Connection keyConnection = DriverManager.getConnection(getUrl(), getUser(), getPass())) {
+                            DatabaseMetaData databaseMetaData = keyConnection.getMetaData();
+
+                            try (ResultSet keyMd = databaseMetaData.getImportedKeys(keyConnection.getCatalog(), null, tableName)) {
+                                while (keyMd.next())
+                                {
+                                    String fkTableName = keyMd.getString("FKTABLE_NAME");
+                                    String fkColumnName = keyMd.getString("FKCOLUMN_NAME");
+                                    String pkTableName = keyMd.getString("PKTABLE_NAME");
+                                    String pkColumnName = keyMd.getString("PKCOLUMN_NAME");
+
+                                    Column fk = null;
+                                    Table fkTable = tableFutures.get(fkTableName).get();
+                                    for (Column c : fkTable.getColumns())
+                                        if (c.getName().equalsIgnoreCase(fkColumnName))
+                                            fk = c;
+                                    if (fk == null)
+                                        throw new RuntimeException("fk column not found: " + fkTableName + "." + fkColumnName);
+                                   Table pkTable = tableFutures.get(pkTableName).get();
+                                    for (Column pk : pkTable.getColumns())
+                                        if (pk.getName().equalsIgnoreCase(pkColumnName))
+                                            fk.setPk(pk);
+
+                                    //log.info(fkTableName + "." + fkColumnName + " -> " + pkTableName + "." + pkColumnName);
+                                }
+                                log.info("{} foreign key processing for table {}", getType(), tableName);
+                                return null;
+                            } catch (ExecutionException e) {
+                                throw new RuntimeException(e);
+                            } catch (InterruptedException e) {
+                                throw new RuntimeException(e);
+                            }
+                        } catch (SQLException ex) {
+                            throw new RuntimeException(ex);
+                        }
+                    }, dbMetadataExecutorPool));
+                }
             }
-            indexMd.close();
 
+         for (CompletableFuture<Table> tableCompletableFuture : tableFutures.values()) {
+            tableCompletableFuture.get();
```

Comment:
There's no point in starting the app if all this isn't set up first, so we force a get on all of the tables and key futures right at the end.